src/Editor/GUI/Editors/EntityEditor.cpp

```diff
@@ -355,65 +356,127 @@ void EntityEditor::RigidBodyEditor(Component::RigidBody* rigidBody) {
-                script->scriptFile = scriptFile;
-        }
+		for (ScriptFile* scriptFile : Hymn().scripts) {
+			if (ImGui::Selectable(scriptFile->name.c_str()))
+				script->scriptFile = scriptFile;
+		}
 
-        ImGui::EndPopup();
-    }
+		ImGui::EndPopup();
+	}
 
-    if (script->scriptFile != nullptr)
-    {
-        ImGui::Text(script->scriptFile->name.c_str());
+	if (script->scriptFile != nullptr)
+	{
+		ImGui::Text(script->scriptFile->name.c_str());
         ImGui::Separator();
 
-        ImGui::Text("Entity References");
-        // Display current entity references
-        for (size_t i = 0; i != script->refList.size(); ++i) {
-            ImGui::Text(script->refList[i]->name.c_str());
-            ImGui::SameLine(ImGui::GetWindowWidth() - 30);
-            if (ImGui::SmallButton(("x###remove" + std::to_string(i)).c_str())) {
-                script->refList.erase(script->refList.begin() + i);
-                break;
-            }
-        }
-        
+        if (ImGui::Button("Fetch properties")) {
 
-        // Choosing other entity references
-        if (ImGui::Button("Add entity reference")) {
-            ImGui::OpenPopup("Add entity reference");
+            Managers().scriptManager->FillPropertyMap(script);
+                
         }
 
-        if (ImGui::BeginPopup("Add entity reference"))
-        {
-            ImGui::Text("Entities");
-            ImGui::Separator();
-            for (Entity* entity : Hymn().world.GetEntities()) //Change into a prettier tree structure or something, later.
-            {
-                if (ImGui::Selectable(entity->name.c_str()))
+        if (script->instance != nullptr) {
+
+            int propertyCount = script->instance->GetPropertyCount();
+
+            for (int n = 0; n < propertyCount; n++) {
+
+                int typeId = script->instance->GetPropertyTypeId(n);
+                void *varPointer = script->instance->GetAddressOfProperty(n);
+                if (typeId == asTYPEID_INT32)
+                {
+                    ImGui::InputInt(script->instance->GetPropertyName(n), (int*)script->propertyMap[script->instance->GetPropertyName(n)][typeId], 0.0f);
+                }
+                else if (typeId == asTYPEID_FLOAT)
+                {
+                    ImGui::DraggableFloat(script->instance->GetPropertyName(n), *(float*)script->propertyMap[script->instance->GetPropertyName(n)][typeId], 0.0f);
+                }
+                else if (typeId & asTYPEID_SCRIPTOBJECT)
                 {
-                    script->refList.push_back(entity);
+                    asIScriptObject *obj = (asIScriptObject*)varPointer;
                 }
+                else if (typeId == script->instance->GetEngine()->GetTypeIdByDecl("string"))
+                {
+                    std::string *str = (std::string*)script->propertyMap[script->instance->GetPropertyName(n)][typeId];
+                    if (str) {
+
+                        if (str->size() <= 128) {
```

Comment:
This is arbitrary. If there is a reason why 128 is a limit, please write a comment that explains it.